### PR TITLE
Introduce with_auth_token to send custom Authorization header

### PR DESCRIPTION
**Pull request by dmitry-silchenko**
Allows to sign request with non-Bearer tokens.

### CODE REVIEW

File: lib/shark.rb [hunk 1]
```diff
@@ -34,26 +34,45 @@ def self.configure
...
+  # @param block [Block] The block where authorization token will be set for
+  # @api public
+  def self.with_auth_token(token)
+    if token.is_a?(String)
+      self.auth_token = token
+    elsif token.respond_to?(:jwt)
+      self.auth_token = token.jwt
+    else
+      raise ArgumentError, 'Parameter :token must be kind of String.'
+    end
```
Comment (jdahlke) [inline lib/shark.rb:61]: Would it not be better / simpler to only allow strings?
What case do you have in mind for `elsif token.respond_to?(:jwt)`?

File: lib/shark.rb [hunk 2]
```diff
@@ -34,26 +34,45 @@ def self.configure
...
+      self.auth_token = "Bearer #{token.jwt}"
     else
       raise ArgumentError, 'Parameter :token must be kind of String.'
     end
 
     yield
   ensure
-    self.service_token = nil
+    self.auth_token = nil
+  end
```
Comment (jdahlke) [inline lib/shark.rb:47]: I suggest to remove code duplication and make use of your more general `#with_auth_token` function here.

```
  def self.with_service_token(token)
    if token.is_a?(String)
      auth_token = "Bearer #{token}"
    elsif token.respond_to?(:jwt)
      auth_token = "Bearer #{token.jwt}"
    end

    with_auth_token(auth_token)
  end
```

**Review by jdahlke — CHANGES_REQUESTED** [review]
Please also add an example to README.

File: lib/shark.rb (hunk 1 above)
Comment (dmitry-silchenko) [reply lib/shark.rb:61]: it was a copy of your code (see `with_service_token`), I thought you know :)

Comment (dmitry-silchenko) [reply lib/shark.rb:61]: I feel I understood, it is for tokens stored in DB. Removing it.

**Review by jdahlke — APPROVED** [review]

**Review by jdahlke — APPROVED** [review]

**Pull request merged**